### PR TITLE
Stag Beetle + Variants

### PR DESCRIPTION
**Pull request by slimeboy460**
#### Purpose of change

I saw in projects that someone wanted to make a beetle which rolls around dead zombies and ate them. I'm hoping this PR can not only add that, but variants and a potential nest area for them. Although, as time goes on the PR might be changed to host different things involving the beetles.

#### Describe the solution

This feature will add a new insect, The Stag Beetle. This beetle (I hope) can get a decent amount of variants and interest.

#### Testing

I have tested it using the game. Where I have already found a few problems which I hope to cover in this PR.

#### Potential Additons
Stag Beetle Larva
Female Stag Beetle (They have a different pincer type. Could be interesting to add.)
Stag Beetle

Flesh eating variants 
 Beetle Wrestler: Has a bigger pair of pincers for better grabbing and throwing at the cost of being too top heavy to fly. Larger and much more aggressive.
Beetle Roller: Rolls around a ball of corpses. (This will probably lag the hell out of people's computers if we make this individual items.)

### CODE REVIEW

**Comment by slimeboy460** [comment]
I just had the idea that the beetles could do a "technical" pulp of the zombies by having a variant which eats them like dead fruit, their sucker mouth thing. Would this be good?

File: data/json/monsters/insect_spider.json [hunk 1]
```diff
@@ -1788,5 +1788,36 @@
     "death_function": [ "NORMAL" ],
     "special_attacks": [ { "type": "leap", "cooldown": 4, "max_range": 4, "allow_no_target": true }, [ "EAT_CROP", 120 ] ],
     "flags": [ "SEES", "HEARS", "SMELLS", "CLIMBS", "LARVA", "STUMBLES", "PATH_AVOID_FIRE" ]
+  },
+  {
+      "id": "mon_stag_beetle",
```
Comment (BrettDong) [inline data/json/monsters/insect_spider.json:1793]:
```suggestion
    "id": "mon_stag_beetle",
```

File: data/json/monsters/insect_spider.json [hunk 2]
```diff
@@ -1788,5 +1788,36 @@
...
+    "type": "MONSTER",
+    "name": { "str": "Huge Stag Beetle" },
+    "description": "Once a priceless bug, it has grown to the size of a calf. Although you doubt that it's going to attack you, those large pincers could probably break your leg.",
```
Comment (BrettDong) [inline data/json/monsters/insect_spider.json:1796]:
```suggestion
    "description": "Once a priceless bug, it has grown to the size of a calf.  Although you doubt that it's going to attack you, those large pincers could probably break your leg.",
```

**Comment by slimeboy460** [comment]
I need to update aggression and special attacks, time to go code diving.

**Comment by Venera3** [comment]
Lovely, beetles are woefully underrepresented and forest are practically bug-free at the moment!
If you manage to get them eating corpses could you do it in a way that I can get my grubby JSON hands on it? Predators murdering things and just leaving the corpse intact is pretty jarring.

File: data/json/monsters/insect_spider.json [hunk 3]
```diff
@@ -1788,5 +1788,36 @@
...
+    "name": { "str": "Huge Stag Beetle" },
+    "description": "Once a priceless bug, it has grown to the size of a calf.  Although you doubt that it's going to attack you, those large pincers could probably break your leg.",
+    "default_faction": "stag_beetle",
+    "bodytype": "flying insect",
+    "species": [ "INSECT" ],
+    "volume": "62500 ml",
+    "weight": "81500 g",
+    "hp": 50,
+    "speed": 110,
+    "material": [ "flesh" ],
```
Comment (Venera3) [inline data/json/monsters/insect_spider.json:1804]:
```suggestion
    "material": [ "iflesh" ],
```
Insects need `"iflesh"` to produce the proper gibs and for insecticide to work on them.

File: data/json/monsters/insect_spider.json [hunk 4]
```diff
@@ -1788,5 +1788,36 @@
...
+    "melee_skill": 4,
+    "melee_dice": 1,
+    "melee_dice_sides": 8,
+    "armor_bash": 12,
+    "armor_cut": 14,
+    "armor_bullet": 10,
+    "dodge": 1,
+    "melee_cut": 0,
+    "vision_day": 10,
+    "harvest": "mammal",
```
Comment (Venera3) [inline data/json/monsters/insect_spider.json:1818]:
```suggestion
    "harvest": "arachnid",
```
Unless you want to add a custom harvest group, the generic arachnid fits them well enough.

File: data/json/monsters/insect_spider.json [hunk 5]
```diff
@@ -1788,5 +1788,36 @@
...
+    "death_function": [ "NORMAL" ],
+    "special_attacks": [ { "type": "grab", "cooldown": 4 }, [ "EAT_CROP", 180 ] ],
```
Comment (Venera3) [inline data/json/monsters/insect_spider.json:1820]:
```suggestion
    "special_attacks": [ [ "GRAB" , 4 ], [ "EAT_CROP", 180 ] ],
```
Grabbing attacks use the old special syntax.

**Review by Venera3 — CHANGES_REQUESTED** [review]

**Comment by slimeboy460** [comment]
Alright, thanks for the advice man! I'm not sure how I could simulate the consumption of corpses. My main concern would be that they downplay the entire threat of the undead coming back, but we could probably just make the time between meals longer.

File: data/json/monsters/insect_spider.json [hunk 6]
```diff
@@ -1788,5 +1788,36 @@
...
+    "melee_dice_sides": 8,
+    "armor_bash": 12,
+    "armor_cut": 14,
+    "armor_bullet": 10,
+    "dodge": 1,
+    "melee_cut": 0,
+    "vision_day": 10,
+    "harvest": "arachnid",
+    "death_function": [ "NORMAL" ],
+    "special_attacks": [ [ "GRAB" , 4 ], [ "EAT_CROP", 180 ] ],
```
Comment (actual-nh) [inline data/json/monsters/insect_spider.json:1820]:
```suggestion
    "special_attacks": [ [ "GRAB", 4 ], [ "EAT_CROP", 180 ] ],
```

File: data/json/monsters/insect_spider.json [hunk 7]
```diff
@@ -1788,5 +1788,36 @@
     "death_function": [ "NORMAL" ],
     "special_attacks": [ { "type": "leap", "cooldown": 4, "max_range": 4, "allow_no_target": true }, [ "EAT_CROP", 120 ] ],
     "flags": [ "SEES", "HEARS", "SMELLS", "CLIMBS", "LARVA", "STUMBLES", "PATH_AVOID_FIRE" ]
+  },
+  {
+    "id": "mon_stag_beetle",
+    "type": "MONSTER",
+    "name": { "str": "Huge Stag Beetle" },
+    "description": "Once a priceless bug, it has grown to the size of a calf.  Although you doubt that it's going to attack you, those large pincers could probably break your leg.",
```
Comment (actual-nh) [inline data/json/monsters/insect_spider.json:1796]: Priceless?

**Review by actual-nh — COMMENTED** [review]
Nice!

**Comment by actual-nh** [comment]
> Alright, thanks for the advice man! I'm not sure how I could simulate the consumption of corpses. My main concern would be that they downplay the entire threat of the undead coming back, but we could probably just make the time between meals longer.

Perhaps a special attack, similar to EAT_FOOD or EAT_CROP?

File: data/json/monsters/insect_spider.json (hunk 7 above)
Comment (slimeboy460) [reply data/json/monsters/insect_spider.json:1796]: Could you explain in a bit more detail what you're trying to ask? I'm a bit tired

**Comment by slimeboy460** [comment]
> > Alright, thanks for the advice man! I'm not sure how I could simulate the consumption of corpses. My main concern would be that they downplay the entire threat of the undead coming back, but we could probably just make the time between meals longer.
> 
> Perhaps a special attack, similar to EAT_FOOD or EAT_CROP?

Me and Venera discussed it, and the way we could probably do this is by making a special attack which reduces the weight and size of the corpse, then labels it with Partially Eaten, Eaten, or just outright removing the corpse/replacing it with bones/waste. This would probably be left to things like flies, wasps and meat eating beetles.

File: data/json/monsters/insect_spider.json (hunk 7 above)
Comment (actual-nh) [reply data/json/monsters/insect_spider.json:1796]: "Once a priceless bug" in the description? (Understand on tired!)

Comment (slimeboy460) [reply data/json/monsters/insect_spider.json:1796]: Once a priceless bug refers to the fact the stag beetle used to be highly sought after pre-cdda for various reasons. One of which was their ability to help keep gardens healthy. Or just because collectors are collectors. 
Tl;DR: It's mainly there for fluff

Comment (actual-nh) [reply data/json/monsters/insect_spider.json:1796]: I partially did the laugh emoji because - re TL;DR - I practically never will **_not_** have read the entire posting, BTW.

Comment (slimeboy460) [reply data/json/monsters/insect_spider.json:1796]: All good lol

**Comment by int-ua** [comment]
Remove the `SUMMARY:` part in the summary line, the format was updated

**Comment by slimeboy460** [comment]
Okay, my C:DDA seems to no longer be glitching, so itt’s time for more play testing!

**Comment by int-ua** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/.github/CONTRIBUTING.md#all-prs-should-have-a-summary-section-with-one-line

**Comment by Night-Pryanik** [comment]
@slimeboy460 is this being worked on?

**Comment by slimeboy460** [comment]
I have been busy with college and relationship stuff, I'd like to get back into devving near Christmas but unfortunately, I've hit kind of a rut due to both of these things. I am assuming you wanna take over?

**Comment by Night-Pryanik** [comment]
No, I'm not planning to take this PR over, just clearing up the queue of PRs in case they're stalled.

**Comment by I-am-Erk** [comment]
Closing as stale, but feel free to reopen when life permits!
